tests/jballerina-unit-test/src/test/resources/test-src/expressions/typecast/type_cast_expr.bal

```diff
@@ -378,7 +378,28 @@ function testFutureCastPositive() returns boolean {
 function testFutureCastNegative() {
     future<int> s1 = start testFutureFunc();
     any a = s1;
-    future<int[]> s2 = <future<int[]>> a;
+    future<int[]>|error s2 = trap <future<int[]>> a;
+
+    assertEquality(true, s2 is error);
+    error err = <error> s2;
+    assertEquality("{ballerina}TypeCastError", err.message());
+    assertEquality("incompatible types: 'future<int>' cannot be cast to 'future<int[]>'", <string> checkpanic err.detail()["message"]);
+}
+
+function foo() returns future<int> {
```

Comment:
Shall we move this function after the call to the function? This will help improve readability (increases vertical affinity)

Reply:
Addressed with the commit 6164d67